bug(python):get_finest_resolution getter and function missmatch

### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the latest version.

### Reproducible Example

```bash
tests/integration/test_collection.py:62:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
ih_muse_k8s/collector.py:29: in collect_and_send_metrics
    interval_td = self.muse.get_finest_resolution().to_timedelta()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <ih_muse.muse.muse.Muse object at 0x10c1db8d0>

    def get_finest_resolution(self) -> TimestampResolution:
        """Get the finest resolution stored by the Poet.

        :return:
            The current TimestampResolution stored in the poet.
        """
>       return self._muse.get_finest_resolution()
E       AttributeError: 'builtins.PyMuse' object has no attribute 'get_finest_resolution'

.venv/lib/python3.11/site-packages/ih_muse/muse/muse.py:80: AttributeError
```


### Bug Description

The get_finest_resolution is a getter in pyo3 but a function in the python_oh_muse

### Version

0.0.23

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -14,7 +14,7 @@ default-members = [
 ]
 
 [workspace.package]
-version = "0.0.23"
+version = "0.0.24"
 edition = "2021"
 rust-version = "1.78"
 homepage = "https://infinitehaiku.com"
@@ -23,12 +23,12 @@ authors = ["Luis Diaz <code.luis.diaz@proton.me"]
 license = "MIT"
 
 [workspace.dependencies]
-ih-muse = { version = "0.0.23", path = "crates/ih-muse" }
-ih-muse-client = { version = "0.0.23", path = "crates/ih-muse-client" }
-ih-muse-core = { version = "0.0.23", path = "crates/ih-muse-core" }
-ih-muse-python = { version = "0.0.23", path = "crates/ih-muse-python" }
-ih-muse-proto = { version = "0.0.23", path = "crates/ih-muse-proto" }
-ih-muse-record = { version = "0.0.23", path = "crates/ih-muse-record" }
+ih-muse = { version = "0.0.24", path = "crates/ih-muse" }
+ih-muse-client = { version = "0.0.24", path = "crates/ih-muse-client" }
+ih-muse-core = { version = "0.0.24", path = "crates/ih-muse-core" }
+ih-muse-python = { version = "0.0.24", path = "crates/ih-muse-python" }
+ih-muse-proto = { version = "0.0.24", path = "crates/ih-muse-proto" }
+ih-muse-record = { version = "0.0.24", path = "crates/ih-muse-record" }
 
 async-trait = "0.1"
 chrono = "0.4.38"

--- a/crates/ih-muse-python/src/muse/general.rs
+++ b/crates/ih-muse-python/src/muse/general.rs
@@ -56,7 +56,7 @@ impl PyMuse {
     }
 
     #[getter]
-    pub fn get_finest_resolution(&self) -> PyTimestampResolution {
+    pub fn finest_resolution(&self) -> PyTimestampResolution {
         let muse = self.muse.clone();
         // Use a blocking mutex lock instead of creating an async context
         let muse_guard = muse.blocking_lock();

--- a/py-ih-muse/ih_muse/ih_muse.pyi
+++ b/py-ih-muse/ih_muse/ih_muse.pyi
@@ -7,8 +7,8 @@ __version__: str
 
 class PyMuse:
     is_initialized: bool
+    finest_resolution: TimestampResolution
     async def initialize(self, timeout: Optional[float] = None) -> None: ...
-    def get_finest_resolution(self) -> TimestampResolution: ...
     def get_remote_element_id(self, local_elem_id: int) -> Optional[int]: ...
     async def register_element(
         self,

--- a/py-ih-muse/ih_muse/muse/muse.py
+++ b/py-ih-muse/ih_muse/muse/muse.py
@@ -63,6 +63,7 @@ class Muse:
         await instance.initialize(timeout)
         return instance
 
+    @property
     def is_initialized(self) -> bool:
         """Check whether the Muse client is initialized.
 
@@ -71,13 +72,14 @@ class Muse:
         """
         return self._muse.is_initialized
 
-    def get_finest_resolution(self) -> TimestampResolution:
+    @property
+    def finest_resolution(self) -> TimestampResolution:
         """Get the finest resolution stored by the Poet.
 
         :return:
             The current TimestampResolution stored in the poet.
         """
-        return self._muse.get_finest_resolution()
+        return self._muse.finest_resolution
 
     def get_remote_element_id(self, local_elem_id: int) -> Optional[int]:
         """Retrieve the remote Element ID associated with a local Element ID.

--- a/tests/test_init.py
+++ b/tests/test_init.py
@@ -11,7 +11,7 @@ from common import MuseTestContext
 async def test_muse_initialization_with_poet() -> None:
     """Test Muse initialization using the Poet client."""
     ctx = await MuseTestContext.create()
-    assert ctx.muse.is_initialized(), "Muse failed to initialize with Poet client"
+    assert ctx.muse.is_initialized, "Muse failed to initialize with Poet client"
 
     local_elem_id = await ctx.register_test_element()
     await ctx.muse.send_metric(local_elem_id, "cpu_usage", 50.0)

--- a/tests/test_init_timeout.py
+++ b/tests/test_init_timeout.py
@@ -47,5 +47,5 @@ async def test_muse_initialization_timeout() -> None:
         await muse.initialize(timeout=0.25)
 
     assert (
-        not muse.is_initialized()
+        not muse.is_initialized
     ), "Muse should not initialize with unreachable endpoint"

--- a/tests/test_muse.py
+++ b/tests/test_muse.py
@@ -21,17 +21,22 @@ async def test_muse() -> None:
         "Metric1",
         "Test Metric",
     )
-
+    default_resolution = ih_muse.TimestampResolution.Seconds
     config = ih_muse.Config(
         endpoints=["http://localhost:8000"],
         client_type=get_client_type_from_env(),
         recording_enabled=False,
         recording_path=None,
-        default_resolution=ih_muse.TimestampResolution.Seconds,
+        default_resolution=default_resolution,
         element_kinds=[element_kind],
         metric_definitions=[metric_definition],
         max_reg_elem_retries=5,
     )
 
     muse = ih_muse.Muse(config)
+
+    assert muse.finest_resolution == default_resolution
+
+    assert not muse.is_initialized
     await muse.initialize(10)
+    assert muse.is_initialized

